stubs/pymacaroons/pymacaroons/macaroon.pyi

```diff
@@ -0,0 +1,49 @@
+from collections.abc import Collection
+
+from pymacaroons import Caveat
+from pymacaroons.binders import HashSignaturesBinder
+from pymacaroons.caveat_delegates import FirstPartyCaveatDelegate, ThirdPartyCaveatDelegate
+from pymacaroons.serializers import BaseSerializer, JsonSerializer
+
+MACAROON_V1: int
+MACAROON_V2: int
+
+class Macaroon:
+    caveats: Collection[Caveat]
+    first_party_caveat_delegate: FirstPartyCaveatDelegate
+    third_party_caveat_delegate: ThirdPartyCaveatDelegate
+    def __init__(
+        self,
+        location: str | bytes | None = None,
+        identifier: str | bytes | None = None,
+        key: str | bytes | None = None,
+        caveats: str | None = None,
+        signature: str | None = None,
+        version: int = 1,
+    ) -> None: ...
+    @classmethod
+    def deserialize(
+        cls, serialized: str | bytes | bytearray, serializer: BaseSerializer | JsonSerializer | None = None
+    ) -> Macaroon: ...
+    @property
+    def location(self) -> str: ...
+    @property
+    def version(self) -> int: ...
+    @property
+    def identifier(self) -> str | bytes: ...
```

Comment:
I'd like advice on what to do with this. A v1 macaroon gets this as `str`, a v2 as `bytes` (set by the `version` parameter to `__init__`).

Reply:
Maybe it would make sense to make the `Macaroon` object generic over the identifier type and add overloads to `__init__`:

```python
class Macaroon(Generic[AnyStr]):
    @overload
    def __init__(self: Self[str], ..., version: Literal[1] = 1) -> None: ...
    @overload
    def __init__(self: Self[bytes], ..., version: Literal[2]) -> None: ...  # all extra arguments without default
    @overload
    def __init__(self: Self[bytes], ..., *, version: Literal[2]) -> None: ...  # all extra arguments with default
    ...
    @property
    def identifier(self) -> AnyStr: ...
```